src/main/java/org/kohsuke/github/GHRelease.java

```diff
@@ -249,15 +251,44 @@ public GHAsset uploadAsset(String filename, InputStream stream, String contentTy
     }
 
     /**
-     * Gets assets.
+     * Get the cached assets.
+     *
+     * @return the assets
+     *
+     * @deprecated This should be the default behavior of {@link #getAssets()} in a future release. This method is
+     *             introduced in addition to enable a transition to using cached asset information while keeping the
+     *             existing logic in place for backwards compatibility.
+     */
+    @Deprecated
+    @Preview
+    public List<GHAsset> getCachedAssets() {
+        return assets;
+    }
+
+    /**
+     * Re-fetch the assets of this release.
      *
      * @return the assets
      * @throws IOException
      *             the io exception
+     * @deprecated The behavior of this method will change in a future release. It will then provide cached assets as
+     *             provided by {@link #getCachedAssets()}. Use {@link #fetchAssets()} instead to fetch up-to-date
+     *             information of assets.
      */
+    @Deprecated
     public List<GHAsset> getAssets() throws IOException {
-        Requester builder = owner.root.createRequest();
+        return fetchAssets();
+    }
 
+    /**
+     * Re-fetch the assets of this release.
+     *
+     * @return the assets
+     * @throws IOException
+     *             the io exception
+     */
+    public List<GHAsset> fetchAssets() throws IOException {
```

Comment:
I left this in as I think this _is_ a quite handy featuer to have (easily run queries to be up to date, if required). If you would rather see this removed please let me know. In that case, the test cases should also be revisitied to cover the same use case with just the new, cached behavior.